host-controller/src/main/java/org/jboss/as/host/controller/HostControllerEnvironmentWrapper.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jboss.as.host.controller;
+
+/**
+ * @author wangc
+ *
+ */
+public class HostControllerEnvironmentWrapper {
+
+    enum HostControllerEnvironmentStatus {
+        NORMAL, // expected abort
+        ERROR // problematic abort
+    }
+
+    private HostControllerEnvironment hostControllerEnvironment;
+    private HostControllerEnvironmentStatus hostControllerEnvironmentStatus;
+
+    private HostControllerEnvironmentWrapper(HostControllerEnvironment hostControllerEnvironment, HostControllerEnvironmentStatus hostControllerEnvironmentStatus) {
+        this.hostControllerEnvironment = hostControllerEnvironment;
+        this.hostControllerEnvironmentStatus = hostControllerEnvironmentStatus;
+    }
+
+    protected HostControllerEnvironmentWrapper(HostControllerEnvironment hostControllerEnvironment) {
+        this(hostControllerEnvironment, null);
+    }
+
+    protected HostControllerEnvironmentWrapper(HostControllerEnvironmentStatus hostControllerEnvironmentStatus) {
```

Comment:
These constructors can be package-protected. There is no subclassing allowed. The classes can even be final.

Sorry to be a pain about this but I kind of have a mania about visibility. It's easier to maintain stuff when visibility is restricted as much as possible; no one then has to guess whether broader visibility was intentional.